### DISCUSSION TITLE
How define config.xml file on react/capacitor project ?

### DISCUSSION THREAD

**Discussion by enes-get-local** [timbru31/cordova-plugin-lottie-splashscreen#390] [Q&A] [answered]
I m using capacitor 3.3.3 with react. Installed this plugin and it shows a white square on splashscreen. show and hide functions works. My problem is that i can't figure out where should put config.xml file ? Ios folder has config.xml file but it is reproducing on every build. It's content like that : 

<?xml version='1.0' encoding='utf-8'?>
<widget version="1.0.0" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">
  <access origin="*" />
  
  <feature name="IonicCordovaCommon">
    <param name="ios-package" value="IonicCordovaCommon" onload="true"/>
  </feature>

  <feature name="LottieSplashScreen">
    <param name="ios-package" value="LottieSplashScreen"/>
    <param name="onload" value="true"/>
  </feature>

  
  <preference name="DisableDeploy" value="true" />
</widget>

Do you have any idea ?

**Comment by timbru31** [comment, marked as answer]
It’s a shame on Ionic’s end that it’s not very good documented. You need to adjust your capacitor JSON config file like in this example: https://capacitorjs.com/docs/cordova/migrating-from-cordova-to-capacitor#cordova-plugin-preferences

**Reply by enes-get-local** [reply]
Thank you for quick response !

I m not sure that i understand correctly. But i added some properties to capacitor.config.json file. Please see config file: 

`{
  "appId": "appId",
  "appName": "appName",
  "bundledWebRuntime": false,
  "npmClient": "npm",
  "webDir": "out",
  "plugins": {
    "LottieSplashScreen": {
      "LottieAnimationLocation": "SplashScreen.json"
    },
    "PushNotifications": {
      "presentationOptions": ["badge", "sound", "alert"]
    },
    "Keyboard": {
      "resize": "none",
      "resizeOnFullScreen": false
    }
  },
  "cordova": {
    "preferences": {
      "DisableDeploy": "true",
      "LottieAnimationLocation": "SplashScreen.json",
      "LottieEnableHardwareAcceleration": "true",
      "LottieFullScreen": "true"
    }
  }
}`

After `npx cap sync` command, config.xml file produced like that: 


<img width="786" alt="Screen Shot 2022-01-08 at 17 13 04" src="https://user-images.githubusercontent.com/70728456/148647417-4e02170d-5867-49dd-a0d1-6b8ed6fa43fe.png">

Do you think where i m doing wrong ?

**Reply by timbru31** [reply]
This looks correct, at least your preferences were correctly picked up by Capacitor.

**Reply by enes-get-local** [reply]
I tried to using remote url and it works. So i guess my local file is can not be found. 
<img width="538" alt="Screen Shot 2022-01-08 at 17 27 56" src="https://user-images.githubusercontent.com/70728456/148647899-cf384b1f-0a61-4e7b-b3af-4154d18ce9f4.png">

As you can see SplashScreen.json file is at the sample place with config.xml file. But it doesn't work

**Reply by enes-get-local** [reply]
UPDATE:

i've put it correct root. it was my fault

**Comment by enes-get-local** [comment]
@timbru31  It works but something is weird. animation is placed at righ bottom side very small while opening the app. Afer it is open a white square is showing up here until hide() function worked.

This is white square: 
<img width="461" alt="Screen Shot 2022-01-08 at 18 31 12" src="https://user-images.githubusercontent.com/70728456/148650123-e91f946b-4540-4210-b442-1a85d2d3de36.png">

Here you can see my animation shows at right bottom


<img width="754" alt="Screen Shot 2022-01-08 at 18 34 33" src="https://user-images.githubusercontent.com/70728456/148650175-4385aae3-62f2-4f35-8803-23fd624cec25.png">

**Reply by enes-get-local** [reply]
@timbru31  Do you have any idea for that ?

config xml
<img width="738" alt="Screen Shot 2022-01-08 at 19 18 27" src="https://user-images.githubusercontent.com/70728456/148651564-77e1ed42-6324-4e24-b758-1a5a9decf51a.png">

**Reply by timbru31** [reply]
Can you by any chance share your project? That would help me to debug the issue.

**Reply by enes-get-local** [reply]
@timbru31  I will see if can add you as contributor to repository but first let me explain as simple

Basically we have a next.js project and have ios/android folders which produced via capacitor init command. Everyting is simple steps on documentations.

I have installed `"cordova-plugin-lottie-splashscreen": "^0.9.6"`.  via yarn.
Configured capacitor.config.json file accordingly documentation and synced ios/android project via `npx cap sync` command.
Here. you can see config.xml file which is produced at ios project :

<img width="1495" alt="Screen Shot 2022-01-10 at 13 10 07" src="https://user-images.githubusercontent.com/70728456/148748948-3817040f-274f-4db4-95d1-8b8832f2222e.png">

Those are all step which i did. So the problem is that animation not centered and not full screen while opening the app. Weird thing is that if i run  `window.lottie.splashscreen.show()` command on console after app runned, it works very well. Problem is on initial start of application.

Also i tested anroid on emulator and it crashes the app.

Please let me know if that is enough or need more information

**Reply by antonioberetini** [reply]
@timbru31 you been added to our repository
